### ISSUE TITLE
Extend function distance with support for n-dimensional input

### ISSUE THREAD

**Issue by jadbox** [josdejong/mathjs#1675]
Currently the distance function only works for a list of arrays with length between 2 and 3. Ideally math.distance should be able to work on arrays of size 1 (just an X cord) or n-dimensional arrays (provided all list of array items are of the same n length).

**Comment by jadbox** [comment]
For n-dim, see: https://github.com/zeke/euclidean-distance

**Comment by josdejong** [comment]
Ah, that would be nice indeed! I think once implemented this n-dimensional implementation could replace the current implementation for 2 and 3 dimensions and make the function as a whole much more concise.

Anyone interested in extending function `distance` to support n-dimensional input?

**Comment by josdejong** [comment]
Ah, that would be nice indeed! I think once implemented this n-dimensional implementation could replace the current implementation for 2 and 3 dimensions and make the function as a whole much more concise.

Anyone interested in extending function `distance` to support n-dimensional input?

**Comment by Veeloxfire** [comment]
how about
```
  function _distanceNd (x, y) {
    const vectorSize = x.length
    let result = 0
    let diff = 0
    for (let i = 0; i < vectorSize; i++) {
      diff = subtract(x[i], y[i])
      result += multiplyScalar(diff, diff)
    }
    return sqrt(result)
  }
```
and
```
  function _Nd (a) {
    // checks if the number of arguments are correct in count and are valid (should be numbers)
    if (a.constructor !== Array) {
      a = _objectToArray(a)
    }
    // If element is not a number return false
    for (let i = 0; i < a.length; i++) {
      if (!_isNumber(a[i])) {
        return false
      }
    }
    // If all elements are numbers return true
    return true
  }
```

only written it for arrays though as proof of concept

**Comment by josdejong** [comment]
Yes, that should do the trick! Would you like to work this out in a PR @Veeloxfire?

**Comment by josdejong** [comment]
Ah I should read my inbox backwards I think :joy: now I see #1702

**Comment by josdejong** [comment]
n-dimensional input is now available in `mathjs@6.4.0`. Thanks again @Veeloxfire
